design/secrets_decoding_design.md

```diff
@@ -173,35 +179,63 @@ Unit tests for any helper functions added to assist with decoding secrets.
 Update integration tests in both [K8s secrets mode](https://github.com/cyberark/secrets-provider-for-k8s/tree/main/pkg/secrets/k8s_secrets_storage/provide_conjur_secrets_test.go) and [P2F mode](https://github.com/cyberark/secrets-provider-for-k8s/tree/main/pkg/secrets/pushtofile/provide_conjur_secrets_test.go) to validate that an encoded variable with base64 content-type is decoded.
 
 ### e2e Tests
-At least one e2e test should be added for validating that an encoded secret with valid annotations is provided and decoded to match the expected value.
+In addition to the unit/integration tests, we likely will want five e2e tests to cover the following cases with secrets decoding:
+* K8s Secrets
+* K8s Secrets with rotation
+* Push-to-file
+* Push-to-file with rotation
+* Decoding secrets > 65k characters
 
 ### Audit 
-The solution should produce an audit entry for each secret that was decoded as a result of the secrets configuration.
+The solution should produce an audit entry for each secret that was decoded as a result of the secrets configuration. It should also log errors/warnings on the following conditions:
+
+Failure on configuration:
+  - Malformed secret (alias/path)
+    - Log YAML parsing error
+    - Secrets Provider fails to start
+  - Malformed content-type
+    - If YAML can still be parsed, create a secret with default content type (text)
+    - Log (warning)
+
+Failure on providing secrets:
```

Comment:
Do we already log a request for a non-existent secret or a secret the API key used doesn't have access to? I'd assume so, but we definitely should if we don't already.

Reply:
Yes, both cases log the resulting 404 if I recall